src/commands/serve.rs

```diff
@@ -356,6 +357,37 @@ impl hyper::service::Service<Request> for ProxyHandler {
         // TODO: need to track the join handle, but don't want to block the response on it
         tokio::task::spawn(async move {
             let req_id = inner.next_req_id();
+            let (mut parts, body) = req.into_parts();
+
+            parts.uri = {
+                let uri_parts = parts.uri.into_parts();
+
+                let scheme = uri_parts.scheme.unwrap_or(http::uri::Scheme::HTTP);
```

Comment:
Why not default to `HTTPS`? Is this also to align with Viceroy?

Reply:
`wasmtime serve` doesn't currently support listening on https, but if that changes in the future we could make it default to what the server's listening on.